integration-test/src/test/java/org/jboss/pnc/integration/BuildTest.java

```diff
@@ -264,25 +270,103 @@ public void shouldBuildTemporaryBuildAndNotAssignItToMilestone() {
         assertThat(buildRecordRestResponse.getValue().getProductMilestoneId()).isNull();
     }
 
+    @Test
+    public void shouldNotTriggerANewPersistentBuildWithoutForceIfOnlyDescriptionChanged() {
+        BuildConfigurationRest buildConfigurationRest = buildConfigurationRestClient.getByName("maven-plugin-test").getValue();
+        BuildOptions persistent = new BuildOptions();
+        persistent.setRebuildMode(RebuildMode.FORCE);
+
+        // Trigger force build
+        RestResponse<BuildRecordRest> forcedPersistentBuild = triggerBCBuild(buildConfigurationRest, Optional.empty(), persistent);
+        assertThat(forcedPersistentBuild.getRestCallResponse().getStatusCode()).isEqualTo(200);
+        ResponseUtils.waitSynchronouslyFor(() -> {
+            RestResponse<BuildRecordRest> record = buildRecordRestClient.get(forcedPersistentBuild.getValue().getId(),
+                    false);
+            return record.hasValue() && record.getValue().getStatus().equals(BuildCoordinationStatus.DONE);
+        } , 15, TimeUnit.SECONDS);
+
+        // Update only description, should not create a new revision and keep same lastModificationTime
+        String oldDescription = buildConfigurationRest.getDescription() != null ? new String(buildConfigurationRest.getDescription()) : null;
```

Comment:
why do you call `new String` here?

Reply:
Why isn't this enough? Strings are immutable
```suggestion
        String oldDescription = buildConfigurationRest.getDescription();
```

Reply:
They are enough, this is leftover of junk code I put when I was frustrated by tests not passing (due to other issues), which I did not remove. Thanks!